DataStructures.Tests/DoublyLinkedListTests.cs

```diff
@@ -0,0 +1,117 @@
+using System;
+using System.Linq;
+using DataStructures.DoublyLinkedList;
+using NUnit.Framework;
+
+namespace DataStructures.Tests {
+    public static class DoublyLinkedListTests {
+        [Test]
+        public static void TestGetData() {
+            var dll = new DoublyLinkedList<int>(new int[] { 0, 1, 2, 3, 4 });
+            var arr = dll.GetData().ToArray();
+            Assert.AreEqual(dll.Count, 5);
+            Assert.AreEqual(new int[] { 0, 1, 2, 3, 4 }, arr);
+        }
+
+        [Test]
+        public static void TestGetAt() {
+            var dll = new DoublyLinkedList<int>(new int[] { 0, 1, 2, 3, 4 });
+            
+            var one = dll.GetAt(1);
+            var three = dll.GetAt(3);
+           
+            Assert.AreEqual(one.Data, 1);
+            Assert.AreEqual(three.Data, 3);
+            Assert.Throws<ArgumentOutOfRangeException>(
+                () =>  dll.GetAt(-1)
+            );
+            Assert.Throws<ArgumentOutOfRangeException>(
+                () =>  dll.GetAt(5)
+            );
+        }
+
+        [Test]
+        public static void TestAddtion() {
+            var dll = new DoublyLinkedList<int>(0);
+            var one = dll.Add(1);
+            dll.Add(3);
+            dll.AddAfter(2, one);
+            dll.Add(4);
+
+            var arr = dll.GetData().ToArray();
+
+            Assert.AreEqual(dll.Count, 5);
+            Assert.AreEqual(new int[] { 0, 1, 2, 3, 4 }, arr);
+        }
+
+        [Test]
+        public static void TestRemove() {
+            var dll = new DoublyLinkedList<int>(new int[] { 0, 1, 2, 3, 4 });
+            dll.RemoveNode(dll.Find(2));
+            dll.RemoveHead();
+            dll.Remove();
+
+            var arr = dll.GetData().ToArray();
+
+            Assert.AreEqual(dll.Count, 2);
+            Assert.AreEqual(new int[] { 1, 3 }, arr);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove the array type; it is redundant.](https://app.codacy.com/manual/siriak/C-Sharp/pullRequest?prid=4803417)